common/src/main/java/io/pravega/common/lang/BigLong.java

```diff
@@ -0,0 +1,78 @@
+/**
+ * Copyright (c) 2017 Dell Inc., or its subsidiaries. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ */
+package io.pravega.common.lang;
+
+import com.google.common.base.Preconditions;
+import io.pravega.common.util.BitConverter;
+import lombok.Data;
+
+@Data
+/**
+ * This class represents a 96 bit number with 32 bit msb encoded as integer and 64 bit lsb encoded as long.
+ * It is unsigned and only allows for non negative values. It also implements comparable interface and comparison involves
+ * first compariging msbs and if msbs are equal then we compare lsbs.
+ */
+public class BigLong implements Comparable {
```

Comment:
Do we have to mark this class as not thread safe?

Reply:
This is an immutable class. So there is no need to mark it as NotThreadSafe or otherwise.